test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_neuron_hosting.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import os
+
+import numpy as np
+import json
+import pytest
+import sagemaker
+from sagemaker.pytorch import PyTorchModel
+from sagemaker.serializers import IdentitySerializer
+from sagemaker.deserializers import BytesDeserializer
+
+from ...integration import model_neuron_dir, resnet_neuron_script, resnet_neuron_input, resnet_neuron_image_list
+from ...integration.sagemaker.timeout import timeout_and_delete_endpoint
+from .... import invoke_pytorch_helper_function
+
+
+@pytest.mark.model("resnet")
+@pytest.mark.processor("neuron")
+@pytest.mark.neuron_test
+def test_neuron_hosting(framework_version, ecr_image, instance_type, sagemaker_regions):
```

Comment:
Could you confirm that this test doesn't run when the image supplied is a non-neuron image or the instance type is non `inf1.*`?

Reply:
Tested them locally and looks like they are skipped

```
(dlcneuron) ubuntu@ip-172-31-34-187:~/dlc/deep-learning-containers/test/sagemaker_tests/pytorch/inference$ python3 -m pytest -v in
tegration/sagemaker/test_neuron_hosting.py --region $REGION --docker-base-name pytorch-inference-cpu --tag 1.9.0-cpu-py38-ubuntu20
.04-2021-09-30-00-35-41-pre-push --framework-version 1.9.0 --processor cpu --py-version 3 --aws-id 647554078242 --instance-type "m
l.c4.8xlarge" --sagemaker-region $REGION
====================================================== test session starts =======================================================
platform linux -- Python 3.6.9, pytest-5.3.5, py-1.10.0, pluggy-0.13.1 -- /home/ubuntu/dlc/deep-learning-containers/dlcneuron/bin/
python3
cachedir: .pytest_cache
rootdir: /home/ubuntu/dlc/deep-learning-containers/test/sagemaker_tests/pytorch/inference
plugins: requests-mock-1.7.0, rerunfailures-9.0, xdist-1.34.0, cov-2.12.1, forked-1.3.0
collected 1 item                                                                                                                 

integration/sagemaker/test_neuron_hosting.py::test_neuron_hosting SKIPPED                                                  [100%]
```

```
(dlcneuron) ubuntu@ip-172-31-34-187:~/dlc/deep-learning-containers/test/sagemaker_tests/pytorch/inference$ python3 -m pytest -v integration/sagemaker/test_mnist.py --region $REGION --docker-base-name pytorch-inference-neuron --tag 1.8.1-neuron-py37-ubuntu18.04-2021-09-29-18-18-42 --framework-version 1.8.1 --processor neuron --py-version 3 --aws-id 647554078242 --instance-type "ml.inf1.xlarge" --sagemaker-region $REGION
====================================================== test session starts =======================================================
platform linux -- Python 3.6.9, pytest-5.3.5, py-1.10.0, pluggy-0.13.1 -- /home/ubuntu/dlc/deep-learning-containers/dlcneuron/bin/python3
cachedir: .pytest_cache
rootdir: /home/ubuntu/dlc/deep-learning-containers/test/sagemaker_tests/pytorch/inference
plugins: requests-mock-1.7.0, rerunfailures-9.0, xdist-1.34.0, cov-2.12.1, forked-1.3.0
collected 3 items                                                                                                                

integration/sagemaker/test_mnist.py::test_mnist_distributed_cpu SKIPPED                                                    [ 33%]
integration/sagemaker/test_mnist.py::test_mnist_distributed_gpu SKIPPED                                                    [ 66%]
integration/sagemaker/test_mnist.py::test_mnist_eia SKIPPED                                                                [100%]
```